app/services/content_sanitizer.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+# Sanitizes and cleans up user-provided content that may contain tags, special characters, etc.
+
+class ContentSanitizer
+  include ActionView::Helpers::SanitizeHelper
+
+  ALLOWED_TAGS = ["p", "b", "strong", "em", "i", "a", "u"].freeze
+  ALLOWED_ATTRIBUTES = ["href", "target"].freeze
+  FILTERED_CHARACTERS = {
+    "&amp;amp;" => "&",
+    "&amp;" => "&",
+    "&nbsp;" => " "
+  }.freeze
+
+  def strip_content(content)
+    content = strip_tags(content.to_s.strip)
+
+    filter_characters(content) if content.length
+  end
+
+  def sanitize_content(content)
+    content = sanitize(content.to_s, tags: ALLOWED_TAGS, attributes: ALLOWED_ATTRIBUTES)
+
+    filter_characters(content) if content.length
+  end
+
+  private
+
+  def filter_characters(content)
+    FILTERED_CHARACTERS.each do |character, sub|
+      content = content.gsub(character, sub)
+    end
+    content
+  end
+end
```

Comment:
I'm missing some tests for this class.

Reply:
67e58257390f8506e92ed964dbf5552f9b5072ca 👍